lexicon/conversion.py

```diff
@@ -72,6 +86,10 @@ def run(self):
                 transform(orth.text) in words
                 or "special" in lemma.attrib
                 or (orth.text is not None and orth.text.startswith("["))
+                or (
+                    self.check_synt_tok
+                    and transform(lemma.find("./synt/tok").text) in words
+                )
                 for orth in lemma.findall("orth")
             ):
```

Comment:
This code lives in an tuple comprehension that runs over all orth of the lemma i.e. is executed multiple times if there are multiple lemmata. A better place would be outside of this comprehension, i.e.

```suggestion
                transform(orth.text) in words
                or "special" in lemma.attrib
                or (orth.text is not None and orth.text.startswith("["))
                for orth in lemma.findall("orth")
            ) or (
                self.check_synt_tok
                and transform(lemma.find("./synt/tok").text) in words
            ):
```
((please test and black this handwritten suggestion 🙃 ))

at this point it might be easier to understand if we defined a function for all these conditions 🙈